docs/contents/releases.md

```diff
@@ -49,80 +55,246 @@ supported by the `GENERALLY AVAILABLE` release of the controller.
+[github.com/aws-controllers-k8s/s3-controller][s3-ctrl] repository, a Docker
+image with a tag `s3-v1.2.6` would be published to the
+[aws-controllers-k8s/controller][ecr-ack-ctrl] ECR repository.
+
+[s3-ctrl]: https://github.com/aws-controllers-k8s/s3-controller
+[ecr-ack-ctrl]: https://gallery.ecr.aws/aws-controllers-k8s/controller
+
+!!! note
+    Binaries for ACK components are published in our Amazon ECR
+    Public [registry][ecr-ack-ctrl].
+
+For ACK components that have a Helm Chart distributable -- i.e. an ACK service
+controller -- the creation of a new SemVer Git tag on the source code
+repository will trigger automatic building and publishing of a Helm Chart with
+an artifact tag including the SemVer version. For example, a Git tag of
+`v1.2.6` on the [github.com/aws-controllers-k8s/s3-controller][s3-ctrl]
+repository means a Helm chart with a tag `s3-v1.2.6` would be published to the
+[aws-controllers-k8s/chart][ecr-ack-chart] ECR repository.
+
+[ecr-ack-chart]: https://gallery.ecr.aws/aws-controllers-k8s/chart
+
+#### A Word About Dependencies
+
+Each service-specific ACK controller -- e.g. the ElastiCache ACK controller --
+depends on a specific version of the ACK common runtime. This dependency is
+specified in the controller's `go.mod` file.
+
+The ACK code generator that produces Go code for service controllers depends on
+a specific version of the ACK common runtime.
+
+!!! note "dependency between the code generator and common runtime"
+    The ACK code generator depends on the ACK common runtime in a unique way:
+    the Go code that the ACK code generator *produces* adheres to a specific
+    version of the ACK common runtime. Even though no Go code in the ACK code
+    generator actually imports the ACK common runtime, this dependency exists
+    because the Go code produced by the templates inside the code generator imports
+    the ACK common runtime. In order to make this Go code dependency more strict,
+    we have a test package inside the ACK code generator that imports the ACK
+    common runtime. In this way, we're able to include a version-specific
+    dependency line in the ACK code generator's `go.mod` file, thereby allowing
+    Go's module infrastructure to pin the dependency between the code generator and
+    the common runtime.
+
+### Stable Helm Charts
+
+!!! tip
+    We [recommend][recommend-helm] using Helm to install an ACK service
+    controller.
+
+[recommend-helm]: https://aws-controllers-k8s.github.io/community/user-docs/install/#helm-recommended
+
+Some ACK service controllers will have Helm Charts with a
+`$SERVICE-v$MAJOR_VERSION-stable` tag, referred from here out as just a
+"`stable` artifact tag". There will only be one of these tags for the ACK
+service controller **in a major version series**. For example, the `stable`
+artifact tag for the ElastiCache ACK service controller's "v1" major version
+series would be `elasticache-v1-stable`.
```

Comment:
Should we add a note on how they should install this specific version with helm? If they don't specify `--version` they'll get the latest published version

```
helm install elasticcache --generate-name --version "v1-stable"
```

Reply:
Yes, @rothgar that is a good idea. We should add that to the install docs. Can I do that in a future PR?

Reply:
Sure, I don't think it _has_ to live here but we should have it written down somewhere. Seeing this made me realize a gap in `helm upgrade` because if customers don't specify `--version` with upgrade then they would upgrade to the latest released chat which could have breaking changes.

I'm going to investigate how helm can lock versions or only follow stable charts. It may require releasing a chart to separate repositories